File: testjava/src/main/java/com/string/StringOperations.java
package com.string;

import java.util.LinkedHashSet;

public class StringOperations {

    public static void main(String arg[]) {
      //  removeDuplicates();
     //   reverseString();
        removeSpaces();
    }

    private static void removeSpaces() {
        String str = "He ll o Wor ld ";
        char[] cArr = str.toCharArray();
        StringBuilder builder = new StringBuilder();
        for (char value : cArr) {
            if (value == ' ')
                continue;
            builder.append(value);
        }
        System.out.println("Before: "+str+"\nAfter: "+builder.toString());
    }

    private static void reverseString() {
        String str = "Hello World";

        char[] charArr = new char[str.length()];
        for (int i = 0; i < str.length(); i++) {
            charArr[i] = str.charAt(str.length() -1 -i);
            charArr[str.length()-1-i] = str.charAt(i);
        }

        System.out.println("before reverse: "+str+"  After reverse: "+new String(charArr));
    }

    private static void removeDuplicates() {
        String str = "Hellooo Worldw";
        System.out.println("String before removing = " + str);

        char[] mychar = str.toCharArray();
        LinkedHashSet<Character> lhs = new LinkedHashSet<>();
        StringBuilder builder = new StringBuilder();
        for (int i = 0; i < mychar.length; i++) {
            lhs.add(str.charAt(i));
        }
        for (Character ch : lhs) {
            builder.append(ch);
        }

        System.out.println("String after removing = " + builder.toString());
    }

}

File: kotlin_prog/src/main/java/com/zafar/kotlin/strings/StringOperations.kt
package strings


fun main(arg: Array<String>){
    //reverseString();
    //removeSpace();
    removeDuplicate()
    findSubString();
}

private fun findSubString() {
    val str1 = "2.5 MB"
    val str2 = "500.65 KB"
    println("Substring of str1 ${str1.substring(0,str1.length-3)}")
    println("Substring of str1 ${str2.substring(0,str2.length-3)}")
}

private fun removeDuplicate() {
    val string: String = "Hello World Hi Hi"
    val lhs = LinkedHashSet<Char>()
    val builder = StringBuilder();
    for (value in string.toCharArray()){
        lhs.add(value)
    }
    for (value in lhs)
        builder.append(value)

    println("After removing duplilcate ${builder.toString()} from $string")
}

private fun removeSpace() {
    val string: String = "Hello Bhai"
    val charArray: CharArray = string.toCharArray()
    val builder = StringBuilder()
    for (char in charArray){
        if (char == ' ')
            continue
        builder.append(char)
    }

    println("After removing space ${builder.toString()} from $string")
}

private fun reverseString() {
    val string: String = "Hello Bhai"
    println("Reverse of $string is ${string.reversed()}")
}
